src/pages/messages/components/TicketDateIntervalFilter.tsx

```diff
@@ -0,0 +1,93 @@
+import { Box } from '@mui/material';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { TicketSearchParam } from '../../../api/searchApi';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { formatDateStringToISO } from '../../../utils/date-helpers';
+
+const commonDatepickerProps: Partial<DatePickerProps<Date>> = {
+  format: 'dd.MM.yyyy',
+  views: ['year', 'month', 'day'],
+  disableHighlightToday: true,
+  slotProps: {
+    textField: { sx: { maxWidth: '10rem' }, size: 'small' },
+  },
+};
+
+export const TicketDateIntervalFilter = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const maxDate = new Date();
+
+  const selectedDatesParam = searchParams.get(TicketSearchParam.CreatedDate);
+
+  const [startParam, endParam] = selectedDatesParam ? selectedDatesParam.split(',') : [];
+  const [selectedFromDate, setSelectedFromDate] = useState<string | undefined>(startParam ? startParam : '');
+  const [selectedToDate, setSelectedToDate] = useState<string | undefined>(endParam ? endParam : '');
+
+  const onChangeFromDate = (newDate: Date | null) => {
+    const newFromDate = newDate ? formatDateStringToISO(newDate) : undefined;
+    setSelectedFromDate(newFromDate);
+    updateSearchParams(`${newFromDate},${selectedToDate}`);
+  };
+
+  const onChangeToDate = (newDate: Date | null) => {
+    const newToDate = newDate ? formatDateStringToISO(newDate) : undefined;
+    setSelectedToDate(newToDate);
+    updateSearchParams(`${selectedFromDate},${newToDate}`);
+  };
+
+  const updateSearchParams = (newDateParam: string) => {
+    const [fromDate, toDate] = newDateParam.split(',');
+
+    if (isValidDate(fromDate) && isValidDate(toDate)) {
+      searchParams.set(TicketSearchParam.CreatedDate, newDateParam);
+    } else if (isValidDate(fromDate)) {
+      searchParams.set(TicketSearchParam.CreatedDate, fromDate);
+    } else if (isValidDate(toDate)) {
+      searchParams.set(TicketSearchParam.CreatedDate, `,${toDate}`);
+    } else {
+      searchParams.delete(TicketSearchParam.CreatedDate);
+    }
```

Comment:
Trengs dette da? Ser ut som du lager verdien som skal hit i `onChangeXxDate()`, så splitter det, og behandler hver verdi hver for seg, bare for å slå det sammen til det samme som var før 🤔 

```suggestion
    searchParams.set(TicketSearchParam.CreatedDate, newDateParam);
```

Reply:
Ser egentlig ut som denne metoden kan fjernes ved å bare legge dette inn i `onChange`-metodene i stedet:

```
searchParams.set(TicketSearchParam.CreatedDate, newDateParam);
history.push({ search: searchParams.toString() });
```